src/server/routes/__tests__/submissions.integration.spec.js

```diff
@@ -381,4 +382,196 @@ describe('submissions.integration', () => {
+      expect(JSON.parse(result.payload)).toEqual({
+        _embedded: {
+          activities: expect.arrayContaining([
+            {
+              _links: {
+                activity: {
+                  href: expect.stringMatching(
+                    `api/activities/${createdActivity1Id}`
+                  )
+                },
+                catches: {
+                  href: expect.stringMatching(
+                    `api/activities/${createdActivity1Id}/catches`
+                  )
+                },
+                river: {
+                  href: expect.stringMatching(
+                    `api/activities/${createdActivity1Id}/river`
+                  )
+                },
+                self: {
+                  href: expect.stringMatching(
+                    `api/activities/${createdActivity1Id}`
+                  )
+                },
+                smallCatches: {
+                  href: expect.stringMatching(
+                    `api/activities/${createdActivity1Id}/smallCatches`
+                  )
+                },
+                submission: {
+                  href: expect.stringMatching(
+                    `api/activities/${createdActivity1Id}/submission`
+                  )
+                }
+              },
+              createdAt: expect.any(String),
+              daysFishedOther: 10,
+              daysFishedWithMandatoryRelease: 20,
+              id: createdActivity1Id,
+              updatedAt: expect.any(String),
+              version: expect.any(String)
+            },
+            {
+              _links: {
+                activity: {
+                  href: expect.stringMatching(
+                    `api/activities/${createdActivity2Id}`
+                  )
+                },
+                catches: {
+                  href: expect.stringMatching(
+                    `api/activities/${createdActivity2Id}/catches`
+                  )
+                },
+                river: {
+                  href: expect.stringMatching(
+                    `api/activities/${createdActivity2Id}/river`
+                  )
+                },
+                self: {
+                  href: expect.stringMatching(
+                    `api/activities/${createdActivity2Id}`
+                  )
+                },
+                smallCatches: {
+                  href: expect.stringMatching(
+                    `api/activities/${createdActivity2Id}/smallCatches`
+                  )
+                },
+                submission: {
+                  href: expect.stringMatching(
+                    `api/activities/${createdActivity2Id}/submission`
+                  )
+                }
+              },
+              createdAt: expect.any(String),
+              daysFishedOther: 3,
+              daysFishedWithMandatoryRelease: 5,
+              id: createdActivity2Id,
+              updatedAt: expect.any(String),
+              version: expect.any(String)
+            }
```

Comment:
please also have a think about these massive expectation statements and how they might be slimmed down a bit - one idea would be a link expectation generator, as this key is just getting repeated with the activity id injected